kansha/board/comp.py

```diff
@@ -886,6 +898,15 @@ def reload_search(self):
         self.must_reload_search = False
         return self.search(self.last_search)
 
+    def save_as_template(self, shared):
+        data = self.data.to_template(shared)
+        fname = unicodedata.normalize('NFKD', self.data.title).encode('ascii', 'ignore')
+        fname = re.sub('\W+', '_', fname.lower())
+        fname = '%s.%s.btpl' % (security.get_user().username, fname)
```

Comment:
Store all the relations (template UUID / user / board) in the DB. Use the `asset manager` to physically store the template on disk and to affect it its UUID